From 153061ea9d26e57801e8808f7e6856bc0a0ca0f0 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Tue, 3 Mar 2015 11:14:45 +0800
Subject: [PATCH] MLK-10436-02: ARM: imx: Add gate clocks for i.MX6QP
 PRE0/1/2/3

There are four PRE engines embedded in i.MX6Q R2.
The PRE0/1/2/3 gate clocks are derived from the gpu3d_axi clock.
This patch adds the gate clocks.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
Signed-off-by: Bai Ping <b51503@freescale.com>
---
 arch/arm/mach-imx/clk-imx6q.c             | 6 ++++++
 include/dt-bindings/clock/imx6qdl-clock.h | 6 +++++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/clk-imx6q.c b/arch/arm/mach-imx/clk-imx6q.c
index 05c7d5a..76f5125 100644
--- a/arch/arm/mach-imx/clk-imx6q.c
+++ b/arch/arm/mach-imx/clk-imx6q.c
@@ -534,6 +534,12 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
 	clk[IMX6QDL_CLK_EIM_SLOW]     = imx_clk_gate2("eim_slow",      "emi_slow_podf",     base + 0x80, 10);
 	clk[IMX6QDL_CLK_VDO_AXI]      = imx_clk_gate2("vdo_axi",       "vdo_axi_sel",       base + 0x80, 12);
 	clk[IMX6QDL_CLK_VPU_AXI]      = imx_clk_gate2("vpu_axi",       "vpu_axi_podf",      base + 0x80, 14);
+	if (cpu_is_imx6q() && imx_get_soc_revision() == IMX_CHIP_REVISION_2_0) {
+		clk[IMX6QDL_CLK_PRE0] = imx_clk_gate2("pre0",          "gpu3d_axi",         base + 0x80, 16);
+		clk[IMX6QDL_CLK_PRE1] = imx_clk_gate2("pre1",          "gpu3d_axi",         base + 0x80, 18);
+		clk[IMX6QDL_CLK_PRE2] = imx_clk_gate2("pre2",          "gpu3d_axi",         base + 0x80, 20);
+		clk[IMX6QDL_CLK_PRE3] = imx_clk_gate2("pre3",          "gpu3d_axi",         base + 0x80, 22);
+	}
 	clk[IMX6QDL_CLK_CKO1]         = imx_clk_gate("cko1",           "cko1_podf",         base + 0x60, 7);
 	clk[IMX6QDL_CLK_CKO2]         = imx_clk_gate("cko2",           "cko2_podf",         base + 0x60, 24);
 
diff --git a/include/dt-bindings/clock/imx6qdl-clock.h b/include/dt-bindings/clock/imx6qdl-clock.h
index 84c5c97..e7a9ad7 100644
--- a/include/dt-bindings/clock/imx6qdl-clock.h
+++ b/include/dt-bindings/clock/imx6qdl-clock.h
@@ -261,6 +261,10 @@
 #define IMX6QDL_CLK_ECSPI_SEL			252
 #define IMX6QDL_CLK_CAN_SEL			253
 #define IMX6QDL_CLK_MMDC_CH1_AXI_CG		254
-#define IMX6QDL_CLK_END				255
+#define IMX6QDL_CLK_PRE0			255
+#define IMX6QDL_CLK_PRE1			256
+#define IMX6QDL_CLK_PRE2			257
+#define IMX6QDL_CLK_PRE3			258
+#define IMX6QDL_CLK_END				259
  
 #endif /* __DT_BINDINGS_CLOCK_IMX6QDL_H */
-- 
1.8.0

